lib/evm_database_ops.rb

```diff
@@ -112,17 +114,18 @@ def self.restore(db_opts, connect_opts = {})
       else
         connect_opts[:remote_file_name] ||= File.basename(backup_file_name(action))
         backup_folder = action == :dump ? "db_dump" : "db_backup"
-        uri = File.join(connect_opts[:uri], backup_folder, connect_opts[:remote_file_name])
+        connect_opts[:uri] = uri = File.join(connect_opts[:uri], backup_folder, connect_opts[:remote_file_name])
       end
 
       session = MiqGenericMountSession.new_session(connect_opts)
       db_opts[:local_file] = session.uri_to_local_path(uri)
+      object_session       = session if session.class.name == "MiqS3Session"
```

Comment:
Why not just yield `session` unconditionally? It can't hurt to provide the mount session for the other types too, right? Then we can remove the extra `object_session` variable entirely.

Reply:
Because the "add" method on the session for backup (and the "download" method on the session for the restore in a separate PR I need to push out from my stash) are only invoked in EvmDatabaseOps if that session variable was yielded.

Reply:
I think it would be more clear to only call add if the session responds to it or check the class when calling the method rather than conditionally yielding. That way the behavior is clear from the `.backup` method rather than having to figure out the cases when the session may not have been yielded.